modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/rule-builder/Calculator/CalculatorButtonArea.es.js

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton, {ClayButtonWithIcon} from '@clayui/button';
+import React from 'react';
+
+import Dropdown from './Dropdown.es';
+
+const ONE_TO_NINE = ['1', '2', '3', '4', '5', '6', '7', '8', '9'];
+
+const OPERATORS = {
+	ADD: {label: Liferay.Language.get('add'), value: '+'},
+	DIVIDE: {label: Liferay.Language.get('divide'), value: '/'},
+	MULTIPLY: {label: Liferay.Language.get('multiply'), value: '*'},
+	SUBTRACT: {label: Liferay.Language.get('subtract'), value: '-'},
+};
+
+const TOKEN_TYPES = {
+	BACKSPACE: 'Backspace',
+	LITERAL: 'Literal',
+	OPERATOR: 'Operator',
+	PARENTHESIS_LEFT: 'Left Parenthesis',
+	PARENTHESIS_RIGHT: 'Parenthesis Right',
+};
+
+function CalculatorButtonArea({
+	disableDot,
+	disableFunctions,
+	disableNumbers,
+	disableOperators,
+	functions,
+	onCalculatorInput,
```

Comment:
When writing this name of property(`onCalculatorInput`) I was planning to add some keyboard functionalities on calculator but I didn't make it work.

However `onClick` is a bit weird, short and too generic for a custom callback, isn't? What do you think about `onCalculatorButtonClick`?

Reply:
You have to simplify the names a bit, they can be short because the name of the component helps in the context of what this API will do, you already have a `CalculatorButtonArea` and `onClick` is a standard buttons API, so there is no a problem in adopting this. Adding `Calculator` is a redundancy in the name of the APIs since your component already has this same prefix.

- This component will not be reused elsewhere
- Component to be used only in a specific place this adds more context

Reply:
Makes sense ;)